shared/test/test_net_sim_api.rb

```diff
@@ -71,6 +81,50 @@ def test_create_read_update_delete
     assert read_records.first.nil?, 'Table was not empty'
   end
 
+  def test_distribute_reads
+    master = MiniTest::Mock.new
+    slave1 = MiniTest::Mock.new
+    slave2 = MiniTest::Mock.new
+    test_redis = Redis::SlaveRead::Interface::Hiredis.new(master: master, slaves: [slave1, slave2])
+    NetSimApi.override_redis_for_test(test_redis)
+
+    # Operations distribute evenly across all three nodes
+    master.expect :hgetall, {}, [String]
+    slave1.expect :hgetall, {}, [String]
+    slave2.expect :hgetall, {}, [String]
+
+    @net_sim_api.get "/v3/netsim/#{@shard_id}/n/1"
+    @net_sim_api.get "/v3/netsim/#{@shard_id}/n/1"
+    @net_sim_api.get "/v3/netsim/#{@shard_id}/n/1"
+    master.verify
+    slave1.verify
+    slave2.verify
+  end
+
+  def test_not_distribute_writes
+    master = MiniTest::Mock.new
+    slave1 = MiniTest::Mock.new
+    slave2 = MiniTest::Mock.new
+    test_redis = Redis::SlaveRead::Interface::Hiredis.new(master: master, slaves: [slave1, slave2])
+    NetSimApi.override_redis_for_test(test_redis)
+
+    # All operations go to master node
+    master.expect :hincrby, 1, [String, String, Fixnum]
+    master.expect :multi, []
+    master.expect :hincrby, 1, [String, String, Fixnum]
+    master.expect :multi, []
+    master.expect :hincrby, 1, [String, String, Fixnum]
+    master.expect :multi, []
+
+    create_record([{name: 'alice', age: 7, male: false}])
+    create_record([{name: 'alice', age: 7, male: false}])
+    create_record([{name: 'alice', age: 7, male: false}])
+
+    master.verify
+    slave1.verify
+    slave2.verify
```

Comment:
Yes; Minitest mocks must be told exactly what they will receive, and complain if they receive any more or less than that. In this case, if a slave received _any_ message you'd get an error like this:
```
NoMethodError: unmocked method :hincrby, expected one of []
```